File: mobile/types/price.ts
export type PriceClassification = "cheap" | "normal" | "expensive";

export interface PriceItem {
  datetime_utc: string;
  value_mwh: number;
  value_kwh: number;
  provider: string;
  classification: PriceClassification;
}

export interface DayPrices {
  date: string;
  provider: string;
  prices: PriceItem[];
  cheapest_hour: PriceItem;
  most_expensive_hour: PriceItem;
}

export interface CurrentPrice {
  datetime_utc: string;
  value_kwh: number;
  provider: string;
  classification: PriceClassification;
}

File: backend/app/schemas/price.py
from pydantic import BaseModel


class PriceItem(BaseModel):
    datetime_utc: str
    value_mwh: float
    value_kwh: float
    provider: str
    classification: str


class CurrentPrice(BaseModel):
    datetime_utc: str
    value_kwh: float
    provider: str
    classification: str


class DayPrices(BaseModel):
    date: str
    provider: str
    prices: list[PriceItem]
    cheapest_hour: PriceItem
    most_expensive_hour: PriceItem
